tooling/noir_js/src/program.ts

```diff
@@ -0,0 +1,20 @@
+/* eslint-disable  @typescript-eslint/no-explicit-any */
+import { Backend } from './backend/backend_interface.js';
+import { generateWitness } from './witness_generation.js';
+
+export class Program {
+  constructor(
+    private circuit: { bytecode: string; abi: any },
+    private backend: Backend,
+  ) {}
+
+  // Initial inputs to your program
+  async generateFinalProof(inputs: any): Promise<Uint8Array> {
```

Comment:
I believe on regular basis I would just make proof, on specific situations I would generate `intermediate` proof?
```suggestion
  async generateProof(inputs: any): Promise<Uint8Array> {
```

Reply:
@signorecello and @Savio-Sou had agreed to make it generateIntermediateProof and generateFinalProof for now